wasm/src/main/java/com/dylibso/chicory/wasm/types/ValueType.java

```diff
@@ -175,20 +222,101 @@ public static int sizeOf(List<ValueType> args) {
         return total;
     }
 
+    @Override
+    public int hashCode() {
+        return Objects.hash(opcode, operand);
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (!(other instanceof ValueType)) {
+            return false;
+        }
+        ValueType that = (ValueType) other;
+        return this.opcode == that.opcode && this.operand == that.operand;
+    }
+
+    public String toString() {
+        switch (opcode) {
+            case Ref:
+            case RefNull:
+                return opcode.name() + "[" + operand + "]";
+            default:
+                return opcode.name();
+        }
+    }
+
+    /**
+     * a string representation of [ValueType] that follows JVM's naming conventions
+     */
+    public String name() {
+        return opcode.name();
+    }
+
     /**
-     * A separate holder class for ID constants.
-     * This is necessary because enum constants are initialized before normal fields, so any reference to an ID constant
-     * in the same class would be considered an invalid forward reference.
+     * basically ValueTypeOpcode, but removes FuncRef and ExternRef, which alias Ref/RefNull
      */
-    static final class ID {
```

Comment:
I think is possible to keep this class here, it will be better for backward compat.